internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -865,3 +877,93 @@ func (provisioner *KopsProvisioner) RefreshKopsMetadata(cluster *model.Cluster)
 
 	return nil
 }
+
+func cleanupMetricsServerResources(provisioner *KopsProvisioner, cluster *model.Cluster) error {
+
+	ctx := context.TODO()
+	logger := provisioner.logger.WithField("cluster", cluster.ID)
+	kops, err := kops.New(provisioner.s3StateStore, logger)
+	if err != nil {
+		return errors.Wrap(err, "failed to create kops wrapper")
+	}
+	defer kops.Close()
+
+	err = kops.ExportKubecfg(cluster.ProvisionerMetadataKops.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to export kubecfg")
+	}
+
+	k8sClient, err := k8s.NewFromFile(kops.GetKubeConfigPath(), logger)
+	if err != nil {
+		return errors.Wrap(err, "failed to construct k8s client")
+	}
```

Comment:
Instead of recreating the k8sClient here, you could just pass it in the function. Same thing with the logger which already has fields set for better logging context.